### ISSUE TITLE
ANNOUNCEMENT: Removal of External Scrapers Support...

### ISSUE THREAD

**Issue by Tikipeter** [Tikipeter/repository.tikipeter#310]
There are a couple of big changes coming to Fen within the next few weeks, beginning with an entirely disruptive change to the way Fen functions.

When installed, Fen has scrapers for Local Folders, Debrid Cloud, Furk and Easynews. These have always been referred to as "Internal Scrapers".

In the past Fen has also supported, and required to be installed, External Scrapers. These are scrapers maintained separately by another dev. But they have been a requirement of installing Fen.

This will now change.

Moving forward, Fen will not require the installation of any external scraper package. It will install as is with only the internal scrapers provided.

If a user wishes to use an external scraper package (or "module"), they will need to install that themselves and instruct Fen to use that scraper module.

The scraper module that currently seems compatible with Fen is the CocoScrapers module. If a user has that module installed already on their system, they can simply direct Fen to use that module by navigating to Settings->Providers->Choose External Scrapers Module.

I assume all users reading this will already have that module installed, so pointing it to Fen will allow it to continue being used.

After this Fen update, there will be an update to the Tikipeter repository removing access to the CocoScrapers Module. The Cocoscrapers dev, or any dev wishing to provide an external scrapers module for use within Fen, will need to provide their own repository so as to provide updates to the module.

I will attempt to get in touch with the CocoScrapers dev with this advice, but after that I will have no interaction with any external scraper module devs in any capacity.

I will also provide zero support for any such external scraper modules. Any issues opened here related to this will be closed, and repeat offenders will be banned.

**Comment by FENLover** [comment]
so sad!! may we ask why?

**Comment by DJMix-I-Can** [comment]
> so sad!! may we ask why?

You don't get it...

**Comment by FENLover** [comment]
> > so sad!! may we ask why?
> 
> You don't get it that this opened up the door for other scrapers...

what other more scrapers good other than coco scrapers? i dont get it sorry

**Comment by umbrellaplug** [comment]
Just wanted to quickly comment here. Umbrella will be doing the same shortly after Fen.

**Comment by DJMix-I-Can** [comment]
> > > so sad!! may we ask why?
> > 
> > 
> > You don't get it that this opened up the door for other scrapers...
> 
>     1. what other more scrapers good other than coco scrapers? i dont get it sorry
> 
>     2. from what repo to install coco scrapers if not come automatic from fen repo
> 
>     3. can multiple external providers be configured? or only one?

You just joined Github about half an hour ago... Re-read what he wrote. He already answered your second & third questions...

**Comment by TwilightMercy** [comment]
> > > > so sad!! may we ask why?
> > > 
> > > 
> > > You don't get it that this opened up the door for other scrapers...
> > 
> > 
> > ```
> > 1. what other more scrapers good other than coco scrapers? i dont get it sorry
> > 
> > 2. from what repo to install coco scrapers if not come automatic from fen repo
> > 
> > 3. can multiple external providers be configured? or only one?
> > ```
> 
> You just joined Github about half an hour ago... Re-read what he wrote. He already answered your second & third questions...

Not the third because I installed fen from test version and it lets only choose one external provider why not multiple

**Comment by HipKat** [comment]
It's a CYA move, I'm sure

**Comment by matthius7** [comment]
It's a protect his arse move and I don't blame him..nothing will change except we have to do a little more foot-work to use Fen. ! Righteous ; at least we aren't left in the dark and to fen for ourselves. Get it Fen for ourselves. 🤣

**Comment by Tikipeter** [comment]
I wanted to keep the comments on this issue open, so thoughtful questions could be asked and answered.

Please don't make me regret it by posting ridiculous, unfounded comments.

This is not reddit.

**Comment by FENLover** [comment]
sorry I will ask more seriously:
In fen settings we can configure only one external scraper? or multiple to use at the same time?

**Comment by Tikipeter** [comment]
Only one. And, at least currently, only Cocoscrapers. I am not expecting this to become some grand ecosystem of scraper modules that Fen can use.

I just need Fen to ship without anything pointing to an external scraper scraping torrent sites. Believe it or not, this is the easiest way to achieve that.

**Comment by racingirl954** [comment]
Thank you Tikipeter for all your hard work and not giving up!! It's most appreciated!!

**Comment by kim1406** [comment]
Understood. Thanks for your hard work. We all appreciate your effort to make FEN one of the best Kodi addons ever made.

**Comment by marcuspax** [comment]
Thanks @Tikipeter, this is the way!

**Comment by Omarscomin** [comment]
Don't panic, guys. This situation is _obviously_ why the "make Fen work properly" setting was created.

**Comment by rizreflects** [comment]
It sounds like a good idea to me.

**Comment by ivarbrandt** [comment]
Makes perfect sense why you are choosing to go this route. Thank you for ensuring the longevity of Fen.

And thank you so much for letting us know.

**Comment by DeFiNiek** [comment]
Thanks for keeping us up to date man!

Very reasonable, now we can enjoy FEN for longer!

**Comment by ozooha** [comment]
The move to distant FEN from external scrapers is the right move.
I am assuming the code related to push and pull from these external scrapers is standardized with CoCo, for now, but down the line (or for other scrapers) we might have to fit things in with FEN?

**Comment by BongSlayer420** [comment]
I just started using Fen a couple weeks ago with Real-Debrid and love it. I just told some aquaintances to try it out but they aren't teach savvy. After the update will the add-on not find results by default?

**Comment by Tikipeter** [comment]
>After the update will the add-on not find results by default?

Correct. The external module for scraping will need to be set within Fen's settings.

**Comment by Tikipeter** [comment]
When Fen next updates, these 5 windows will show giving some info. They will only show once. (Please excuse the window transparency issue as a result of buggy Kodi screen capturing.)

![screenshot00000](https://github.com/Tikipeter/repository.tikipeter/assets/28930002/161011df-07f2-4440-a241-9f0fd68e1888)

![screenshot00001](https://github.com/Tikipeter/repository.tikipeter/assets/28930002/5fde7a4e-dc38-42b6-8acb-ca1943b66abc)

![screenshot00002](https://github.com/Tikipeter/repository.tikipeter/assets/28930002/a8a3a8a6-a9a4-4687-8da6-f1177df66f62)

![screenshot00003](https://github.com/Tikipeter/repository.tikipeter/assets/28930002/7f877c9d-24a6-4984-ba7c-837ddbd608dc)

![screenshot00004](https://github.com/Tikipeter/repository.tikipeter/assets/28930002/27ea3ffe-fc00-4d0a-b6cf-e0aa2a79c2c2)
